testsuite/MDAnalysisTests/coordinates/test_fhiaims.py

```diff
@@ -23,32 +23,159 @@
-
-from MDAnalysisTests.datafiles import FHIAIMS
+from six import StringIO
+from six.moves import zip
+import os
 
 import MDAnalysis as mda
-import os
 import numpy as np
+from MDAnalysisTests import make_Universe
+from MDAnalysisTests.coordinates.base import (
+    _SingleFrameReader, BaseWriterTest)
+from MDAnalysisTests.datafiles import FHIAIMS
+from numpy.testing import (assert_equal,
+                           assert_array_almost_equal,
+                           assert_almost_equal)
+
+
+class RefFHIAIMS(object):
+    from MDAnalysis.coordinates.FHIAIMS import (FHIAIMSReader, FHIAIMSWriter)
+
+    filename, trajectory, topology = [FHIAIMS] * 3
+    reader, writer = FHIAIMSReader, FHIAIMSWriter
+    pos_atom1 = np.asarray(
+        [6.861735,   2.103823,  37.753513], dtype=np.float32)
+    dimensions = np.asarray(
+        [18.6, 18.6,  55.8, 90., 90., 90.], dtype=np.float32)
+    n_atoms = 6
+    n_frames = 1
+    time = 0.0
+    ext = '.in'
+    prec = 6
+    container_format, changing_dimensions = [False] * 2
+
+
+class TestFHIAIMSReader(object):
+
+    prec = 6
+
+    @pytest.fixture
+    def ref(self):
+        return RefFHIAIMS()
+
+    @pytest.fixture
+    def universe(self):
+        return mda.Universe(FHIAIMS, FHIAIMS)
+
+    @pytest.fixture
+    def universe2(self):
+        return mda.Universe(FHIAIMS)
+
+ #   def setUp(self):
+ #       self.ref = RefFHIAIMS
+ #       self.prec = 6
+
+    def test_single_file(self, universe, universe2):
+        assert_almost_equal(universe.atoms.positions, universe2.atoms.positions,
+                            self.prec, "FHIAIMSReader failed to load universe from single file")
+
+    def test_uses_FHIAIMSReader(self, universe):
+        from MDAnalysis.coordinates.FHIAIMS import FHIAIMSReader
+
+        assert isinstance(universe.trajectory,
+                          FHIAIMSReader), "failed to choose FHIAIMSReader"
+
+    def test_dimensions(self, ref, universe):
+        assert_almost_equal(
+            ref.dimensions, universe.dimensions,
+            self.prec, "FHIAIMSReader failed to get unitcell dimensions")
+
+    def test_n_atoms(self, ref, universe):
+        assert_equal(
+            ref.n_atoms, universe.trajectory.n_atoms,
+            "FHIAIMSReader failed to get the right number of atoms")
+
+    def test_fhiaims_positions(self, ref, universe):
+        # first particle
+        assert_almost_equal(ref.pos_atom1,
+                            universe.atoms.positions[0],
+                            self.prec,
+                            "FHIAIMSReader failed to read coordinates properly")
+
+    def test_n_frames(self, ref, universe):
+        assert_equal(ref.n_frames, universe.trajectory.n_frames,
+                     "wrong number of frames")
+
+    def test_time(self, ref, universe):
+        assert_equal(ref.time, universe.trajectory.time,
+                     "wrong time of the frame")
+
+    def test_bad_input1(self, tmpdir):
```

Comment:
Something like this is good.

You can actually create the file as a string buffer with `StringIO` and use it directly as show in https://github.com/MDAnalysis/mdanalysis/blob/0911ff3a0379356ee603ff4aa21eec3b990f9958/testsuite/MDAnalysisTests/coordinates/test_gro.py#L432-L440

I would make it a fixture and then use it somehow like this:
```python
from six import StringIO

@pytest.fixture(scope="module")
def bad_input_missing_periodicity():
    buffer = """
lattice_vector 1.0 0.0 0.0
lattice_vector 0.0 1.0 0.0
atom 0.1 0.1 0.1 H
"""
   return StringIO(buffer)

...

   def test_bad_input1(self, bad_input_missing_periodicity):
        with pytest.raises(ValueError, match="Found partial periodicity"):
             u = mda.Universe(bad_input_missing_periodicity, format="FHIAIMS")
```

Also note the use of `pytest.raises` to check for the correct exception and `match` to check for the error message itself.

Reply:
Excellent, just made it class-wide